src/custom/hooks/useMetadataDoc.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react'
+import { useActiveWeb3React } from 'hooks/web3'
+import { generateReferralMetadataDoc, AppDataDoc } from 'utils/metadata'
+
+export function useMetadataDoc() {
+  const { account } = useActiveWeb3React()
```

Comment:
@alfetopito @matextrem shouldn't this be the address from the url instead of the wallet's address?

Reply:
Good point, you want to use the address of who referred you, not yourself :)

But you do want to do that only when you are connected == here's an `account` value set.